### ISSUE TITLE
errors that occur "between" jobs result in lost jobs

### ISSUE THREAD

**Issue by carlost** [steelThread/mimeograph#39]
this method is in mimeograph.coffee ~ line 703

``` coffee
storeAndEnqueue: (file, encoding..., job, jobId) ->
    options = key: file
    options.encoding = encoding.shift() unless _.isEmpty encoding
    file2redis file, options, (err) =>
      return @capture err, {jobId: jobId, job: job} if err?
      @enqueue job, file, jobId
```

if the callback to file2redis is given an err we call return and log an error message with @capture.  However, the next job is never enqueue and a fatal or pagejob error is never recorded.  Which means that mimeo will never notify the client about the results of this job ... not good.

this situation is not limited to the storeAndEnqueue method. it can occur in a number of spots.  in general if we run into an error after coffee-resque calls success and before we notify/enqueue another job:
- notify will not be called
- the next job will not be enqueued

We need to do a beter job about closing this hole.  As far as i can tell this has been possible for a while (at least since v1.0) however, the toolchain is way more complex and some tools we use (e.g. pdftk and pdfbeads) are fussy.

**Comment by steelThread** [comment]
Do it!

**Comment by carlost** [comment]
closed as per pull request #41
